core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -439,6 +437,14 @@ class KafkaController(val config: KafkaConfig,
     registerBrokerModificationsHandler(newBrokers)
   }
 
+  private def maybeResumeReassignments(shouldResume: ReplicaAssignment => Boolean): Unit = {
+    controllerContext.partitionsBeingReassigned.foreach { tp =>
```

Comment:
The new logic guarantees that the reassignment is stored in ZK before it is added to `partitionsBeingReassigned` - nice!